### ISSUE TITLE
Compilation error when not using optional tiff

### ISSUE THREAD

**Issue by pnorman** [mapnik/mapnik#2625]
./configure reports

```
Note: will build without these OPTIONAL dependencies:
...
   - tiff (TIFF C library | configure with TIFF_LIBS & TIFF_INCLUDES)
...
```
but `make` has the error 
```
++ -o tests/cxx/tiff_io.o -c -std=c++11 -Wall -pthread -ftemplate-depth-300 -Wno-pragmas -O3 -DSHAPE_MEMORY_MAPPED_FILE -DMAPNIK_HAS_DLCFN -DBIGINT -DHAVE_JPEG -DMAPNIK_USE_PROJ4 -DHAVE_PNG -DLINUX -DMAPNIK_THREADSAFE -DBOOST_SPIRIT_NO_PREDEFINED_TERMINALS=1 -DBOOST_PHOENIX_NO_PREDEFINED_TERMINALS=1 -DBOOST_SPIRIT_USE_PHOENIX_V3=1 -DNDEBUG -DSHAPE_MEMORY_MAPPED_FILE -DMAPNIK_HAS_DLCFN -DBIGINT -DHAVE_JPEG -DMAPNIK_USE_PROJ4 -DHAVE_PNG -DLINUX -DMAPNIK_THREADSAFE -DBOOST_SPIRIT_NO_PREDEFINED_TERMINALS=1 -DBOOST_PHOENIX_NO_PREDEFINED_TERMINALS=1 -DBOOST_SPIRIT_USE_PHOENIX_V3=1 -DNDEBUG -DGRID_RENDERER -DHAVE_LIBXML2 -Ideps -Ideps/clipper/include -Ideps/agg/include -Iinclude -I/home/ubuntu/boost/include -I/usr/include -I/usr/include/freetype2 -I/usr/include/libxml2 -I/usr/include/postgresql tests/cxx/tiff_io.cpp
In file included from tests/cxx/tiff_io.cpp:6:0:
include/mapnik/tiff_io.hpp:33:20: fatal error: tiffio.h: No such file or directory
 #include <tiffio.h>
                    ^
compilation terminated.
scons: *** [tests/cxx/tiff_io.o] Error 1
scons: building terminated because of errors.
make: *** [mapnik] Error 2
```

This should reproduce it

```
sudo apt-get update
sudo apt-get dist-upgrade
sudo apt-get --no-install-recommends install git automake autoconf make libtool g++ libcurl4-openssl-dev python-dev \
libicu-dev python-dev libxml2 libxml2-dev libfreetype6 libfreetype6-dev libjpeg-dev libpng-dev libfreetype6 libfreetype6-dev libxml2 libxml2-dev libpq-dev libharfbuzz-dev libproj-dev

mkdir src
# Get sources
curl -L -o boost_1_57_0.tar.gz http://sourceforge.net/projects/boost/files/boost/1.57.0/boost_1_57_0.tar.gz/download
git clone -q https://github.com/mapnik/mapnik.git

tar -zxvf boost_1_57_0.tar.gz
cd boost_1_57_0

mkdir -p $HOME/boost
./bootstrap.sh --with-libraries=program_options,thread,filesystem,python,regex,system --prefix=$HOME/boost
./b2 install

cd ~/src/mapnik
./configure BOOST_INCLUDES=$HOME/boost/include/boost BOOST_LIBS=$HOME/boost/lib PREFIX=$HOME/mapnik BINDINGS=none

**Comment by springmeyer** [comment]
Thanks for catching this, fixed in a96cf38

**Comment by pnorman** [comment]
Confirming that it's fixed.
